qutebrowser/completion/models/listcategory.py

```diff
@@ -0,0 +1,121 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2017 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The base completion model for completion in the command line.
+
+Module attributes:
+    Role: An enum of user defined model roles.
+"""
+
+import re
+
+from PyQt5.QtCore import QSortFilterProxyModel
+from PyQt5.QtGui import QStandardItem, QStandardItemModel
+
+from qutebrowser.utils import qtutils, debug, log
+
+
+class ListCategory(QSortFilterProxyModel):
+
+    """Expose a list of items as a category for the CompletionModel."""
+
+    def __init__(self, name, items, columns_to_filter=None, parent=None):
+        super().__init__(parent)
+        self.name = name
+        self.srcmodel = QStandardItemModel(parent=self)
+        self.pattern = ''
```

Comment:
Used in `lessThan`:_

        leftstart = left.startswith(self.pattern)
        rightstart = right.startswith(self.pattern)